packages/gafl-webapp-service/src/pages/recurring-payments/cancel/details/__tests__/route.spec.js

```diff
@@ -0,0 +1,162 @@
+        expect(pageRoute).toHaveBeenCalledWith(
+          CANCEL_RP_DETAILS.page,
+          CANCEL_RP_DETAILS.uri,
+          expect.any(Function),
+          expect.any(Function),
+          expect.any(Function)
+        )
+      })
+    })
+  })
+
+  it('calls addLanguageCodeToUri with request object', async () => {
+    jest.isolateModules(() => {
+      require('../route.js')
+    })
+    const [[, , , completion]] = pageRoute.mock.calls
+    const sampleRequest = Symbol('sample request')
+    completion(sampleRequest)
+    expect(addLanguageCodeToUri).toHaveBeenCalledWith(sampleRequest, expect.anything())
+  })
+
+  it('calls addLanguageCodeToUri with CANCEL_RP_AUTHENTICATE uri', () => {
+    jest.isolateModules(() => {
+      require('../route.js')
+    })
+    const [[, , , completion]] = pageRoute.mock.calls
+    completion({})
+    expect(addLanguageCodeToUri).toHaveBeenCalledWith(expect.anything(), CANCEL_RP_CONFIRM.uri)
+  })
+
+  it('returns the value of addLanguageCodeToUri', () => {
+    jest.isolateModules(() => {
+      require('../route.js')
+    })
+    const [[, , , completion]] = pageRoute.mock.calls
+    const expectedCompletionRedirect = Symbol('expected-completion-redirect')
+    addLanguageCodeToUri.mockReturnValueOnce(expectedCompletionRedirect)
+
+    const completionRedirect = completion({})
+
+    expect(completionRedirect).toBe(expectedCompletionRedirect)
+  })
+
+  const getSampleCatalog = () => ({
+    rp_cancel_details_licence_holder: 'Licence holder',
+    rp_cancel_details_licence_type: 'Licence type',
+    rp_cancel_details_payment_card: 'Payment card',
+    rp_cancel_details_last_purchased: 'Last purchased',
+    rp_cancel_details_licence_valid_until: 'Valid until'
+  })
+
+  const createMockRequest = ({ currentPermission, catalog = getSampleCatalog() } = {}) => ({
+    cache: () => ({
+      helpers: {
+        transaction: {
+          getCurrentPermission: jest.fn().mockResolvedValue(currentPermission)
+        }
+      }
+    }),
+    i18n: {
+      getCatalog: () => catalog
+    }
+  })
+
+  describe('getData', () => {
+    beforeEach(() => {
+      jest.clearAllMocks()
+    })
+
+    it('returns request.i18n.getCatalog()', async () => {
+      const currentPermission = {
+        permission: {
+          licensee: {
+            firstName: 'John',
+            lastName: 'Smith'
+          },
+          permit: {
+            description: 'Salmon and sea trout'
+          },
+          endDate: '01-01-2026',
+          referenceNumber: 'abc123'
+        },
+        recurringPayment: {
+          lastDigitsCardNumbers: 1234
+        }
+      }
+
+      const mssgs = {
+        rp_cancel_details_licence_holder: 'Test licence holder',
+        rp_cancel_details_licence_type: 'Test licence type',
+        rp_cancel_details_payment_card: 'Test payment card',
+        rp_cancel_details_last_purchased: 'Test last purchased',
+        rp_cancel_details_licence_valid_until: 'Test valid until'
+      }
```

Comment:
This, and the sample permission above, would be better in simple generation functions as they're both copied and pasted into the next test. The functions can be super simple, taking no arguments, as both objects are identical in each test.